dbt/adapters/snowflake/connections.py

```diff
@@ -151,6 +152,18 @@ def auth_args(self):
             result["client_store_temporary_credential"] = True
             # enable mfa token cache for linux
             result["client_request_mfa_token"] = True
+
+        # Warning: This profile configuration can result in specific threads, even just one,
```

Comment:
Context in PR description.

My proposed way of handling what appears to be a rare bug. Warn people against it. Don't encourage it. We could try to manually close threads but setting an explicit timeout seems sketchy to me.